_posts/2024-09-16-introducing-opensearch-2-17.md

```diff
@@ -0,0 +1,109 @@
+
+**Use asynchronous batch ingestion for efficient high-volume ML task processing**
+
+OpenSearch 2.17 also introduces [_asynchronous batch ingestion_](https://opensearch.org/docs/latest/ml-commons-plugin/remote-models/async-batch-ingestion/), which allows you to trigger batch inference jobs, monitor job status, and ingest results once batch processing is complete. This streamlines high-volume machine learning tasks, such as generating embeddings for large datasets and ingesting them into k-NN indexes.
+
+**Customize ML inference search response processors for your use case**
+
+OpenSearch 2.16 introduced ML inference search processors, enabling you to run model predictions while executing search queries. Starting with OpenSearch 2.17, you can now choose to run model predictions for all documents in a single request or run model predictions for each document individually, further enhancing search response processors. This search response processor improvement provides you with the flexibility to customize ML inference search response processors to suit your specific use case.
+
+## Search
+
+This release also delivers new functionality focused on helping you improve search and query performance.
+
+**Optimize search performance with enhanced concurrent segment search settings and expanded support**
+
+[Concurrent segment search](https://opensearch.org/docs/latest/search-plugins/concurrent-segment-search/) has been generally available since OpenSearch 2.12. It supports two tuning parameters: an/off switch and a cluster-level setting that specifies the number of slices (maximum thread count) per request. In general, a greater number of slices reduces latency but requires more processing power per request. [More specific analysis](https://opensearch.org/blog/concurrent-search-follow-up/) has shown that some request types (including aggregations and remote store searches) benefit more from concurrent segment search than others. To help you take advantage of concurrent segment search for the right request types, OpenSearch 2.17 introduces a new setting both at index and cluster level. These settings along with [pluggable _decider_ logic](https://github.com/opensearch-project/OpenSearch/pull/15363) provides more granular control over which requests use concurrent search. This release also supports a [per-index (as opposed to per-cluster) maximum slice count setting](https://github.com/opensearch-project/OpenSearch/pull/15336). Additionally, we have expanded concurrent segment search to support more script-based search requests, including [composite aggregations with scripting](https://github.com/opensearch-project/OpenSearch/pull/15072) and [derived fields](https://github.com/opensearch-project/OpenSearch/pull/15326).
```

Comment:
```suggestion
[Concurrent segment search](https://opensearch.org/docs/latest/search-plugins/concurrent-segment-search/) has been generally available since OpenSearch 2.12. It supports two tuning parameters: an/off switch and a cluster-level setting that specifies the number of slices (maximum thread count) per request. In general, a higher number of slices reduces latency but requires more processing power per request. [More specific analysis](https://opensearch.org/blog/concurrent-search-follow-up/) has shown that some request types (including aggregations and remote store searches) benefit more from concurrent segment search than others. To help you take advantage of concurrent segment search for the right request types, OpenSearch 2.17 introduces a new setting both at the index and cluster levels. These settings, along with [pluggable _decider_ logic](https://github.com/opensearch-project/OpenSearch/pull/15363), provide more granular control over which requests use concurrent search. This release also supports a [per-index (as opposed to per-cluster) maximum slice count setting](https://github.com/opensearch-project/OpenSearch/pull/15336). Additionally, we have expanded concurrent segment search to support more script-based search requests, including [composite aggregations with scripting](https://github.com/opensearch-project/OpenSearch/pull/15072) and [derived fields](https://github.com/opensearch-project/OpenSearch/pull/15326).
```